src/mod/applications/mod_conference/conference_member.c

```diff
@@ -241,6 +241,7 @@ switch_status_t conference_member_add_event_data(conference_member_t *member, sw
 		return status;
 
 	if (member->conference) {
+		conference_utils_set_flag(member->conference, CFLAG_HAS_MEMBER_ID);
```

Comment:
The issue with that approach is that the channel data added in `conference_event_add_data` will only add the first member channel details which is inaccurate for valid events that has member-id to locate their channel details. 

So I only want channels data to be added to events that calls `conference_event_add_data` and not `conference_member_add_event_data`.

Reply:
The flag is to distinguish conference events that has member ID and those that doesn't. So that we only use first member channel data for conference event with no member ID.

Reply:
If that is the case then we can have something like this:
```
switch_status_t conference_event_add_data(conference_obj_t *conference, switch_event_t *event)
{
   conference_event_add_data_with_member(conference, event, NULL);
}

switch_status_t conference_event_add_data_with_member(conference_obj_t *conference, switch_event_t *event, conference_member_t *member)
{
        switch_status_t status = SWITCH_STATUS_SUCCESS;

	if (member) {
		switch_channel_t *channel = switch_core_session_get_channel(member->session);
		if (member->verbose_events) {
			switch_channel_event_set_data(channel, event);
		} else {
			switch_channel_event_set_basic_data(channel, event);
		}
		switch_event_add_header(event, SWITCH_STACK_BOTTOM, "Video", "%s",
								switch_channel_test_flag(switch_core_session_get_channel(member->session), CF_VIDEO) ? "true" : "false" );
	} else {
		conference_member_t *first_member = NULL;

		for (first_member = conference->members; first_member; first_member = first_member->next) {
			if (first_member->session) {
				switch_channel_t *channel = switch_core_session_get_channel(first_member->session);
				switch_channel_event_set_basic_data(channel, event);
				break;
			}
		}
	}

        
	switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "Conference-Name", conference->name);
	switch_event_add_header_string(event, SWITCH_STACK_BOTTOM, "Conference-Domain", conference->domain);
        .... // Same logic with the conference_event_add_data
}

.... // in conference_member_add_event_data

if (member->conference) {
     status = conference_event_add_data_with_member(member->conference, event. member);
		switch_event_add_header(event, SWITCH_STACK_BOTTOM, "Floor", "%s", (member->id == member->conference->floor_holder) ? "true" : "false" );
}

```

Reply:
Good approach. Implemented. Thanks.